app/Services/GraphQL.php

```diff
@@ -15,14 +16,50 @@ public function __construct()
     }
 
     /**
-     * Run a GraphQL query using the client.
+     * Run a GraphQL query using the client
+     * and parse the data result into a convenient format.
      *
      * @param  $query     String
      * @param  $variables Array
-     * @return \Softonic\GraphQL\Response
+     * @param  $queryKey  String
+     * @return array|null
      */
-    public function query($query, $variables)
+    public function query($query, $variables, $queryKey)
```

Comment:
I think I'd recommend against having a `$queryKey` parameter here, since we might very well make queries that include multiple keys (e.g. querying `posts(…) { … } and `user(id: …) { … }` in one shot in the future.

Reply:
@DFurnes do we currently make queries like that anywhere? I'd love to familiarize myself! I would have thought those would have been two separate queries making `$queryKey` okay to use here

Reply:
One that comes to mind is [the petition block](https://github.com/DoSomething/phoenix-next/blob/abecd18eb47ea9bfddd492efb71de275514cec48/resources/assets/components/actions/PetitionSubmissioncAction/PetitionSubmissionAction.js#L33-L42), where we check both for any existing submission & the user's first name for the "signature" field. This is definitely something that is possible (and encouraged) in GraphQL!

Reply:
Ah makes sense! Updated the parsing and removed the `$queryKey` param in https://github.com/DoSomething/rogue/pull/889/commits/45c6850df0efb7d0e867c97002072594b2dbfa24